### PR TITLE
[1.8.9] Some World Mappings

### PR DESCRIPTION
**Pull request by Grayray75**

### CODE REVIEW

File: mappings/net/minecraft/world/PersistentState.mapping [hunk 1]
```diff
@@ -1,4 +1,4 @@
-CLASS net/minecraft/class_625 net/minecraft/world/level/storage/PersistentState
+CLASS net/minecraft/class_625 net/minecraft/world/PersistentState
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/world/PersistentState.mapping:1]: why this change

File: mappings/net/minecraft/world/PersistentStateManager.mapping [hunk 2]
```diff
@@ -1,4 +1,4 @@
-CLASS net/minecraft/class_643 net/minecraft/world/level/storage/PersistentStateManager
+CLASS net/minecraft/class_643 net/minecraft/world/PersistentStateManager
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/world/PersistentStateManager.mapping:1]: ditto

File: mappings/net/minecraft/world/TemperoryStateManager.mapping [hunk 3]
```diff
@@ -0,0 +1,2 @@
+CLASS net/minecraft/class_641 net/minecraft/world/TemperoryStateManager
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/world/TemperoryStateManager.mapping:1]: ditto

File: mappings/net/minecraft/world/PersistentState.mapping (hunk 1 above)
Comment (BoogieMonster1O1) [reply mappings/net/minecraft/world/PersistentState.mapping:1]: wait upstream has it like this
nvm

**Review by BoogieMonster1O1 — APPROVED** [review]

File: mappings/net/minecraft/world/PersistentState.mapping (hunk 1 above)
Comment (Grayray75) [reply mappings/net/minecraft/world/PersistentState.mapping:1]: Because Upstream has it like this

File: mappings/net/minecraft/world/World.mapping [hunk 4]
```diff
@@ -333,8 +393,8 @@ CLASS net/minecraft/class_99 net/minecraft/world/World
 	METHOD method_387 addEntity (Lnet/minecraft/class_1745;)Z
 		ARG 1 entity
 	METHOD method_389 getSkyAngleRadians (F)F
-	METHOD method_391 (Lnet/minecraft/class_646;)Z
-		ARG 1 boundingBox
+	METHOD method_391 doesAreaContainFluid (Lnet/minecraft/class_646;)Z
+		ARG 1 area
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/world/World.mapping:397]: conc why

File: mappings/net/minecraft/world/World.mapping [hunk 5]
```diff
@@ -344,15 +404,20 @@ CLASS net/minecraft/class_99 net/minecraft/world/World
 	METHOD method_394 spawnEntity (Lnet/minecraft/class_1745;)Z
 		ARG 1 entity
 	METHOD method_396 doesAreaContainFireSource (Lnet/minecraft/class_646;)Z
-		ARG 1 boundingbox
+		ARG 1 area
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/world/World.mapping:407]: `Box` as `area` sounds conc

**Review by BoogieMonster1O1 — CHANGES_REQUESTED** [review]
next time i should review it with greater detail

File: mappings/net/minecraft/world/TemperoryStateManager.mapping (hunk 3 above)
Comment (halotroop2288) [inline mappings/net/minecraft/world/TemperoryStateManager.mapping:1]: *Temporary

File: mappings/net/minecraft/world/World.mapping [hunk 6]
```diff
@@ -37,7 +39,7 @@ CLASS net/minecraft/class_99 net/minecraft/world/World
 		ARG 2 properties
 		ARG 3 dim
 		ARG 4 profiler
-		ARG 5 client
+		ARG 5 isClient
```
Comment (halotroop2288) [inline mappings/net/minecraft/world/World.mapping:42]: `is` is a getter prefix.

**Review by halotroop2288 — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/world/World.mapping (hunk 5 above)
Comment (halotroop2288) [reply mappings/net/minecraft/world/World.mapping:407]: What I take more issue in is the long name `doesXContainY` could just be `containsY`.

File: mappings/net/minecraft/world/World.mapping (hunk 6 above)
Comment (BoogieMonster1O1) [reply mappings/net/minecraft/world/World.mapping:42]: its used in fields as well

Comment (BoogieMonster1O1) [reply mappings/net/minecraft/world/World.mapping:42]: no harm in using it for parameters

File: mappings/net/minecraft/world/TemperoryStateManager.mapping (hunk 3 above)
Comment (Grayray75) [reply mappings/net/minecraft/world/TemperoryStateManager.mapping]: fixed

**Review by halotroop2288 — APPROVED** [review]

File: mappings/net/minecraft/world/World.mapping (hunk 5 above)
Comment (Grayray75) [reply mappings/net/minecraft/world/World.mapping:407]: I renamed it to so it's consistent with the others. What wrong about `area`?

Comment (BoogieMonster1O1) [reply mappings/net/minecraft/world/World.mapping:407]: `Box` can be named as `box`. self explanatory

Comment (BoogieMonster1O1) [reply mappings/net/minecraft/world/World.mapping:407]: or `boundingBox`

Comment (BoogieMonster1O1) [reply mappings/net/minecraft/world/World.mapping:407]: please not `area`

Comment (BoogieMonster1O1) [reply mappings/net/minecraft/world/World.mapping:407]: `area` is also a 2 dimensional term
gets confusing

File: mappings/net/minecraft/world/World.mapping (hunk 4 above)
Comment (BoogieMonster1O1) [reply mappings/net/minecraft/world/World.mapping:397]: upstream uses `containsFluid`

File: mappings/net/minecraft/world/World.mapping [hunk 7]
```diff
@@ -125,55 +136,83 @@ CLASS net/minecraft/class_99 net/minecraft/world/World
...
 	METHOD method_286 isInsideWorld (Lnet/minecraft/class_395;Lnet/minecraft/class_1745;)Z
 		ARG 1 border
 		ARG 2 entity
 	METHOD method_288 isRegionLoaded (Lnet/minecraft/class_476;)Z
 		ARG 1 box
 	METHOD method_290 (Lnet/minecraft/class_646;)Ljava/util/List;
 		ARG 1 box
-	METHOD method_291 (Lnet/minecraft/class_646;Lnet/minecraft/class_591;)Z
-		ARG 1 boundingBox
+	METHOD method_291 doesAreaContainMaterial (Lnet/minecraft/class_646;Lnet/minecraft/class_591;)Z
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/world/World.mapping:167]: `containsMaterial`

File: mappings/net/minecraft/world/World.mapping (hunk 5 above)
Comment (halotroop2288) [reply mappings/net/minecraft/world/World.mapping:407]: > `area` is also a 2 dimensional term
> gets confusing

`volume` would be the correct alternative then.

File: mappings/net/minecraft/world/World.mapping [hunk 8]
```diff
@@ -282,15 +334,22 @@ CLASS net/minecraft/class_99 net/minecraft/world/World
 	METHOD method_357 getLightAtPos (Lnet/minecraft/class_106;Lnet/minecraft/class_1372;)I
 		ARG 1 type
 		ARG 2 pos
+	METHOD method_358 addUnloadedBlockEntity (Lnet/minecraft/class_348;)V
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/world/World.mapping:337]: this should probably be `queueBlockEntity`, since the list is cleared and block entities are added in the next tick.

File: mappings/net/minecraft/world/World.mapping [hunk 9]
```diff
@@ -200,18 +241,24 @@ CLASS net/minecraft/class_99 net/minecraft/world/World
 		ARG 1 clazz
 		ARG 2 box
 		ARG 3 entityPredicate
+	METHOD method_319 (Ljava/lang/Class;Lnet/minecraft/class_646;Lnet/minecraft/class_1745;)Lnet/minecraft/class_1745;
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/world/World.mapping:244]: `getEntitiesByClass`
first param is `entityClass`
last param is `except`

File: mappings/net/minecraft/world/World.mapping [hunk 10]
```diff
@@ -125,55 +136,83 @@ CLASS net/minecraft/class_99 net/minecraft/world/World
...
 		ARG 1 pos
 		ARG 2 distance
 	METHOD method_301 isRegionLoaded (Lnet/minecraft/class_1372;IZ)Z
 		ARG 1 pos
 		ARG 2 distance
+		ARG 3 canBeEmpty
 	METHOD method_302 getLightAtPos (Lnet/minecraft/class_1372;Lnet/minecraft/class_106;)I
 		ARG 1 pos
 		ARG 2 type
+	METHOD method_303 (Lnet/minecraft/class_1372;Lnet/minecraft/class_160;)Z
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/world/World.mapping:203]: `hasScheduledTick`

File: mappings/net/minecraft/world/World.mapping [hunk 11]
```diff
@@ -125,55 +136,83 @@ CLASS net/minecraft/class_99 net/minecraft/world/World
...
+		ARG 0 worldView
+		ARG 1 pos
 	METHOD method_281 getLuminance (Lnet/minecraft/class_106;Lnet/minecraft/class_1372;)I
 		ARG 1 type
 		ARG 2 pos
 	METHOD method_282 (Lnet/minecraft/class_106;Lnet/minecraft/class_1372;I)V
 		ARG 1 type
 		ARG 2 pos
 		ARG 3 level
+	METHOD method_283 (Lnet/minecraft/class_160;Lnet/minecraft/class_1372;Ljava/util/Random;)V
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/world/World.mapping:148]: `scheduleTick`

File: mappings/net/minecraft/world/World.mapping [hunk 12]
```diff
@@ -125,55 +136,83 @@ CLASS net/minecraft/class_99 net/minecraft/world/World
 	METHOD method_279 setPropertiesInitialized (Lnet/minecraft/class_102;)V
 		ARG 1 info
 	METHOD method_280 isOpaque (Lnet/minecraft/class_104;Lnet/minecraft/class_1372;)Z
+		ARG 0 worldView
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/world/World.mapping:139]: `world` is fine for this

File: mappings/net/minecraft/world/biome/Biome.mapping [hunk 13]
```diff
@@ -124,6 +138,10 @@ CLASS net/minecraft/class_113 net/minecraft/world/biome/Biome
 			ARG 2 variationModifier
 		METHOD method_548 diminish ()Lnet/minecraft/class_113$class_114;
 	CLASS class_115 Temperature
+		FIELD field_434 OCEAN Lnet/minecraft/class_113$class_115;
+		FIELD field_435 COLD Lnet/minecraft/class_113$class_115;
+		FIELD field_436 MEDIUM Lnet/minecraft/class_113$class_115;
+		FIELD field_437 WARM Lnet/minecraft/class_113$class_115;
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/world/biome/Biome.mapping:144]: these are enum values
they get automatically mapped by stitch
hence they're grey in enigma

**Review by BoogieMonster1O1 — CHANGES_REQUESTED** [review]
please change all `area` names to `box`

File: mappings/net/minecraft/world/World.mapping (hunk 5 above)
Comment (BoogieMonster1O1) [reply mappings/net/minecraft/world/World.mapping:407]: > > `area` is also a 2 dimensional term
> > gets confusing
> 
> `volume` would be the correct alternative then.

doesn't matter

Comment (BoogieMonster1O1) [reply mappings/net/minecraft/world/World.mapping:407]: upstream uses box

Comment (BoogieMonster1O1) [reply mappings/net/minecraft/world/World.mapping:407]: `public <T extends Entity> List<T> getEntitiesByType(@Nullable EntityType<T> type, Box box, Predicate<? super T> predicate)`  
this has box

File: mappings/net/minecraft/world/biome/Biome.mapping (hunk 13 above)
Comment (halotroop2288) [reply mappings/net/minecraft/world/biome/Biome.mapping:144]: And your point is?_

Comment (BoogieMonster1O1) [reply mappings/net/minecraft/world/biome/Biome.mapping:144]: > And your point is?

just saying, they don't need to be mapped

and all comments aren't for requesting changes. they're just comments

File: mappings/net/minecraft/world/World.mapping (hunk 5 above)
Comment (Grayray75) [reply mappings/net/minecraft/world/World.mapping:409]: fixed

File: mappings/net/minecraft/world/World.mapping [hunk 14]
```diff
@@ -125,55 +136,83 @@ CLASS net/minecraft/class_99 net/minecraft/world/World
...
 	METHOD method_290 (Lnet/minecraft/class_646;)Ljava/util/List;
 		ARG 1 box
-	METHOD method_291 (Lnet/minecraft/class_646;Lnet/minecraft/class_591;)Z
-		ARG 1 boundingBox
-	METHOD method_293 hasEntityIn (Lnet/minecraft/class_646;Lnet/minecraft/class_1745;)Z
+	METHOD method_291 containsMaterial (Lnet/minecraft/class_646;Lnet/minecraft/class_591;)Z
 		ARG 1 box
+		ARG 2 material
+	METHOD method_292 (Lnet/minecraft/class_646;Lnet/minecraft/class_591;Lnet/minecraft/class_1745;)Z
+		ARG 1 area
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/world/World.mapping:171]: here

File: mappings/net/minecraft/world/World.mapping [hunk 15]
```diff
@@ -125,55 +136,83 @@ CLASS net/minecraft/class_99 net/minecraft/world/World
...
+		ARG 2 material
+		ARG 3 entity
+	METHOD method_293 hasEntityIn (Lnet/minecraft/class_646;Lnet/minecraft/class_1745;)Z
+		ARG 1 area
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/world/World.mapping:175]: here

File: mappings/net/minecraft/world/World.mapping (hunk 5 above)
Comment (BoogieMonster1O1) [reply mappings/net/minecraft/world/World.mapping:409]: still says area here 🤔

File: mappings/net/minecraft/world/World.mapping [hunk 16]
```diff
@@ -282,15 +336,22 @@ CLASS net/minecraft/class_99 net/minecraft/world/World
...
 		ARG 2 pos
+	METHOD method_358 queueBlockEntity (Lnet/minecraft/class_348;)V
+		ARG 1 blockEntity
 	METHOD method_359 isRegionLoaded (Lnet/minecraft/class_476;Z)Z
 		ARG 1 box
+		ARG 2 canBeEmpty
 	METHOD method_360 hasEntityIn (Lnet/minecraft/class_646;)Z
-		ARG 1 box
-	METHOD method_361 containsBlockWithMaterial (Lnet/minecraft/class_646;Lnet/minecraft/class_591;)Z
 		ARG 1 area
```
Comment (BoogieMonster1O1) [inline mappings/net/minecraft/world/World.mapping:345]: while you're at it, could you replace this as well?

**Review by BoogieMonster1O1 — COMMENTED** [review]
just a few left

File: mappings/net/minecraft/world/World.mapping (hunk 16 above)
Comment (Grayray75) [reply mappings/net/minecraft/world/World.mapping]: all boxes should be fixed now

Comment (BoogieMonster1O1) [reply mappings/net/minecraft/world/World.mapping]: alright thanks

**Review by BoogieMonster1O1 — APPROVED** [review]
LGTM
